Classes/YLRefreshHeaderView/YLRefreshHeaderView.m

```diff
@@ -31,26 +35,37 @@ - (void)setRefreshState:(YLRefreshHeaderViewState)refreshState {
 }
 
 - (void)setRefreshState:(YLRefreshHeaderViewState)refreshState animated:(BOOL)animated {
-  _refreshState = refreshState;
+  if (_refreshState != refreshState) {
+    _refreshState = refreshState;
 
-  void (^animationBlock)() = ^{
-    UIScrollView *strongScrollView = self.scrollView;
-    UIEdgeInsets contentInset = strongScrollView.contentInset;
-    // TODO(mglidden): make this support view controllers that have UIRectEdgeTop
-    contentInset.top = (refreshState == YLRefreshHeaderViewStateRefreshing) ? self.pullAmountToRefresh : 0;
-    strongScrollView.contentInset = contentInset;
-  };
-  void (^completionBlock)(BOOL) = ^(BOOL finished) {
-    if (self.refreshState == YLRefreshHeaderViewStateClosing) {
-      self.refreshState = YLRefreshHeaderViewStateNormal;
+    void (^animationBlock)() = ^{
+      UIScrollView *strongScrollView = self.scrollView;
+      UIEdgeInsets contentInset = strongScrollView.contentInset;
+      switch (refreshState) {
+        case YLRefreshHeaderViewStateRefreshing:
+          self.previousTopInset = contentInset.top;
+          contentInset.top = self.pullAmountToRefresh;
```

Comment:
Ah, I'm guessing the 64px is the height of the navigation bar overlapping the tableview. I'm not sure if this is something I should change, because the previous implementation assumed that contentInset.top is initially zero, resets to zero, and that the top edge of the table view is aligned with the bottom edge of the navigation bar. So if you have a pullAmountToRefresh of 15, the new contentInset.top would be 15 as well. Implementations that use YLTableView may already rely on this behavior.  This would present a bigger challenge when contentInset.top is initially negative, which would mean part of the refresh header would be cut off